src/ripple/app/misc/HashRouter.h

```diff
@@ -107,12 +106,26 @@ class HashRouter
             return true;
         }
 
+        /** Determines if this item should be recovered from the open ledger.
+
+            Counts the number of times the item has been recovered.
+            If it hits the limit, reset the counter and return false.
+            Else, increment the counter and return true.
+
+            @note The limit must be > 0
+        */
+        bool shouldRecover(std::uint32_t limit)
+        {
+            return ++recoveries_ % limit != 0;
```

Comment:
So `getDefaultRecoverLimit` returns `1` and since `x mod 1` is always `0` this will _always_ return `false`. If we make `getDefaultRecoverLimit` return `2`, then this function will oscillate between `true` and `false`. Not sure if that's what we want.

I know that I was the one that suggested this change, but I'm now thinking that this should be reduced to:

```
bool shouldRecover(std::uint32_t limit)
{
    return recoveries_++ < limit;
}
```

With `limit == 1` this function will return to the sequence `true, false, false, false, ...`

Reply:
```
    HashRouter (Stopwatch& clock, std::chrono::seconds entryHoldTimeInSeconds,
        std::uint32_t recoverLimit)
        : suppressionMap_(clock)
        , holdTime_ (entryHoldTimeInSeconds)
        , recoverLimit_ (recoverLimit + 1u)
    {
    }
```
That does the right thing, works after rediscovery, and allows the API to specify "recover this many times before giving up."